### ISSUE TITLE
Closes 10-15 seconds after attaching to Factorio

### ISSUE THREAD

**Issue by Nuttar** [cheat-engine/cheat-engine#3033]
Even without a .ct, if i just open CE and attach to factorio while in the menu then CE just closes shortly after. 

- I've tried: ceregreset.exe
- windowsrepair.exe
- using older CE7.2 i had a backup folder of
- tried removing CE mentions from registry myself
- reinstalling 7.5 fresh
- tried using both VEH Debugging and Windows

I can attach to other games just fine, even with factorio open. Any ideas that I can try?

**Comment by cheat-engine** [comment]
delete windbg from the win64 subfolder

or go to settings->debug settings, and tick the option to not load debug symbols

**Comment by Nuttar** [comment]
I used the option and that did work, but the table did not so I assume it's required to work.
And yea it works fine just leaving it open without attaching. I can get a cheat option or 2 working and staying on before it closes itself, if that matters.

**Comment by cheat-engine** [comment]
yes, delete dbghelp.dll

**Comment by Nuttar** [comment]
Win 10 Pro 22H2.
Tried after deleting dbghelp, with the dbg symbol option unticked and it lasts an extra 10 seconds-ish

**Comment by cheat-engine** [comment]
does the folder of terraria have a dbghelp dll as well? if so cooy that one over to ce's win64 folder and try that

**Comment by Symbai** [comment]
I can attach to Factorio (legit steam version) without a problem so its likely something else and not the game which causes it.

**Comment by Davoc11** [comment]
exactly the same problem. but windows 11. is there any solution?

**Comment by cheat-engine** [comment]
do you have any extra factorio DLC or mods installed?

And the solution seems to be to untick the usage of debug symbols, or delete the .pdb

**Comment by Davoc11** [comment]
only dlc space age
thanks i'll try

**Comment by cheat-engine** [comment]
also, how much RAM do you have and can your pagefile still grow ?

**Comment by Nuttar** [comment]
yea i've got the space age dlc and mods. though i've tried it without mods. i'm guessing it's just another rare problem from software conflicts that'd be hard to track down. i'm considering this a loss to be honest, if you want to close this.

**Comment by Nuttar** [comment]
![image](https://github.com/user-attachments/assets/66bcb720-bbcf-4162-aabe-00bfba177bbd)
got 32gb ram, 15gb used. paging file seems ok

**Comment by Davoc11** [comment]
Thank you very much. It seems to have helped - "untick the usage of debug symbols"

**Comment by Nuttar** [comment]
sure it technically stops SE from crashing. but the cheat table for factorio doesn't function without it.

**Comment by Davoc11** [comment]
Yes, it's a pity that you are right. The tables stop working, although CE is stable.

**Comment by cheat-engine** [comment]
try deleting `%temp%\Cheat Engine Symbols` and then try again.  Perhaps you've got an issue with the file there.

**Comment by Nuttar** [comment]
> try deleting `%temp%\Cheat Engine Symbols` and then try again. Perhaps you've got an issue with the file there.
> 
> I have installed that dlc as well, and tested it but it works fine. Have you guys tried attaching to Factorio without having a table loaded ? (in case having the table loaded is what causes the issue)

Still closes after deleting that folder. and yea i've tried without loading the table, it's how i've tested whether any solution works first.

**Comment by Davoc11** [comment]
the same

**Comment by Davoc11** [comment]
![no name](https://github.com/user-attachments/assets/11a33a72-8a7a-4177-9a99-3a6f7c086c4d)

**Comment by Nuttar** [comment]
Mine climbs to 1gb and 10% cpu till it closes

**Comment by Davoc11** [comment]
I didn't pay attention to it before, but when working, CE eats up all the RAM
![Без імені](https://github.com/user-attachments/assets/f540ca86-494b-4037-b999-fee6d3ee17b8)

**Comment by Davoc11** [comment]
Yes, also usually close the window when it reaches 1 gigabyte. But sometimes it happens that it uses all the RAM

**Comment by cheat-engine** [comment]
That's why i mentioned the pagefile has to be able to grow
It stores all the data in the pagefile for quick lookup later. and if that can't grow then CE will likely terminate

**Comment by Davoc11** [comment]
I don't think that's the issue. I have 4 disks and each one is set to system managed size

**Comment by Davoc11** [comment]
on version CE 7.1 - everything works for me, the tables work. This is of course not a complete solution, but better than nothing

**Comment by Nuttar** [comment]
7.1 doesn't have the option to disable debug symbols either, so i guess it doesn't crash due to a lack of the feature? the table doesn't work for it anyway. 

Ended up looking for new game version table, and they made a non-symbol debug version and references this thread; and it works just fine. So thanks for that, Bloodybone.

If cheat-engine wants to close this if it's considered not quite a bug or otherwise solved, feel free. I don't know if i should, like if it's kept open for bug tracking purposes or not.

**Comment by askdata01** [comment]
Having the same issue only with Factorio.  It was working fine yesterday then stopped today.  I have tried deleting the dbghelp.dll, turning off all Exploit protection in MS security, and disabling antivirus completely. I have tried it with laptop with fresh install of both CE and Factorio.  It will stay active for about 10 - 15 seconds then CE closes with no message.  Windows Event viewer shows the following:
Faulting application name: cheatengine-x86_64-SSE4-AVX2.exe, version: 7.5.0.7431, time stamp: 0x00000000
Faulting module name: ntdll.dll, version: 10.0.26100.2454, time stamp: 0x7cb6b6a8
Exception code: 0xc0000374
Fault offset: 0x00000000000881f5
Faulting process id: 0x3B00
Faulting application start time: 0x1DB4E23ED23D7CF
Faulting application path: C:\Program Files\Cheat Engine 7.5\cheatengine-x86_64-SSE4-AVX2.exe
Faulting module path: C:\WINDOWS\SYSTEM32\ntdll.dll
Report Id: f8bf58dc-6265-4497-93f3-edef2403c028
Faulting package full name: 
Faulting package-relative application ID: 

I've also tried running the non-SSE version with the same results.  I also tried running the 32-bit version of CE and it does not crash but it gives the following error: Scan error: controller: No readable memory found  I wonder if this indicates that Factorio is now loading in upper memory and is protected some how.

I also tried creating a folder in the CE folder for the temp scan files with full permissions for all users in case it couldn't write to the wintemp dir.  Still crashes.

I have checked and there have been no Windows updates, nor Factorio updates between when it worked and when it didn't.  Steam may have updated but VAC shouldn't be an issue since this is a standalone game.

For reference, I am running Windows 11 24H2 and all drivers are up to date.

**Comment by lstor** [comment]
Same issue trying to attach to Door Kickers 2. Disabling loading of the pdb (by renaming it to `.txt`) solved the issue, so it is highly likely related to CE's pdb-handling.

**Comment by cheat-engine** [comment]
you're using ce 7.6 ?

**Comment by lstor** [comment]
Didn't realize there had been an update. No problems in 7.6, cheers!
